### DISCUSSION TITLE
STT is Extremely Slow; is there ways to increase speed?

### DISCUSSION THREAD

**Discussion by dbuchacher** [coqui-ai/STT#2196] [General]
EDIT: I think I found my own answer.  However, any additional input is welcomed.
```shell
for f in *.wav; do ( stt --model model.tflite --scorer huge-vocabulary.scorer --audio ${f} > ${f}.txt ) & done
```

ORIGINAL TEXT:

Umm, the title kinda explains it all.

Background Information:
* Have a folder with a bunch of podcasts mp3 files. (about 150 episodes; between 1 to 2 hours long)
* I converted all mp3 files to `.wav` file format by `mpg123 -w new.wav old.mp3`
* Now I'm looping through all the `.wav` files to transcribe to text files.
* `for f in *.mp3; do stt --model model.tflite --scorer huge-vocabulary.scorer --audio ${f} > ${f}.txt`
* This is working but it's `super fucking slow`. I started it before bed.  Now after waking up I've found only about 25 episodes transcribed.  At this rate, it will literally take days.

Am I doing something wrong?  Or is there a better way to do this?

I've noticed only 1 CPU core is doing anything and the rest of them are sitting idle at 0%.
* can I set it to use more CPUs?

Maybe coqui-ai-stt isn't a good tool for this job?  Is there a better tool I should be using?

Thanks

**Comment by FrontierDK** [comment]
Have you thought of having several VMs / PCs? Using 1 PC to control them all via SSH / bash scripting, you can get the job done quicker...

One of the best things about Linux distros...is that Linux is free ;) And no activation. Also, the Ubuntu server supports dynamic memory in hypervisors, and the server takes up very little memory when not in use...

**Comment by wasertech** [comment]
What you need is multithreading. Using a for loop in a shell script spawns only one transcription process. You could make a socket server to host the model and spawn multiple clients at the same time to do your bidding. Like I did with [Listen](https://gitlab.com/waser-technologies/technologies/listen).
